spec/json.coffee

```diff
@@ -0,0 +1,341 @@
+if typeof process isnt 'undefined' and process.execPath and process.execPath.indexOf('node') isnt -1
+  chai = require 'chai' unless chai
+  parser = require '../lib/fbp'
+  parser.validateSchema = true # validate schema for every test on node.js. Don't have tv4 in the browser build
+else
+  parser = require 'fbp'
+
+describe 'JSON to FBP parser', ->
+  it 'should provide a parse method', ->
+    chai.expect(parser.parse).to.be.a 'function'
+    chai.expect(parser.serialize).to.be.a 'function'
+
+  describe 'roundtrip', ->
+    describe 'with simple FBP string', ->
+      fbpData = """
+      '8003' -> LISTEN WebServer(HTTP/Server) REQUEST -> IN Profiler(HTTP/Profiler) OUT -> IN Authentication(HTTP/BasicAuth)
+      Authentication() OUT -> IN GreetUser(HelloController) OUT[0] -> IN[0] WriteResponse(HTTP/WriteResponse) OUT -> IN Send(HTTP/SendResponse)
+      'hello.jade' -> SOURCE ReadTemplate(ReadFile) OUT -> TEMPLATE Render(Template)
+      GreetUser() DATA -> OPTIONS Render() OUT -> STRING WriteResponse()
+      """
+      roundTrippedFbpData = ""
+      graphString = ""
+      graphData = null
+      graphData2 = null
+
+      it 'should produce a graph JSON object', ->
+        # fbp -> json
+        graphData = parser.parse fbpData, caseSensitive:true
+        chai.expect(graphData).to.be.an 'object'
+        chai.expect(graphData.caseSensitive).to.equal true
+
+        # json -> fbp
+        roundTrippedFbpData = parser.serialize graphData
+
+        # fbp -> json
+        graphData2 = parser.parse roundTrippedFbpData, caseSensitive:true
+
+        # json -> fbp, to test
+        fbp2 = parser.serialize graphData2
+
+        chai.expect(roundTrippedFbpData).to.equal fbp2
+
+      describe 'the generated graph', ->
+        it 'should contain eight nodes', ->
+          chai.expect(graphData2.processes).to.be.an 'object'
+          chai.expect(graphData2.processes).to.have.keys [
+            'WebServer'
+            'Profiler'
+            'Authentication'
+            'GreetUser'
+            'WriteResponse'
+            'Send'
+            'ReadTemplate'
+            'Render'
+          ]
+        it 'should contain ten edges and IIPs', ->
+          chai.expect(graphData2.connections).to.be.an 'array'
+          chai.expect(graphData2.connections.length).to.equal 10
+        it 'should contain no exports', ->
+          chai.expect(graphData2.exports).to.be.an 'undefined'
+
+  describe 'with flowhub json graph (from ingress c-base table)', ->
+    jsonData = """
```

Comment:
You've got a point about the browser tests. Silly that it is so tricky to load a file asset there...